### PR TITLE
Resource scroll displays warehouse contents

### PR DESCRIPTION
**Pull request by Thodor12**
No issue present, discussed with Ray over Discord

# Changes proposed in this pull request:
- When shift right-clicking on a warehouse you can create a "snapshot"
- As long as there is a snapshot loaded on the resource scroll it'll display amounts (just like with deliveries) next to the requested items

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/client/gui/WindowResourceList.java [hunk 1]
```diff
@@ -54,15 +71,45 @@
...
+     * The position of the warehouse clicked on.
+     */
+    @Nullable
+    private final BlockPos warehousePos;
+
+    /**
+     * The compound data to store the warehouse snapshot to.
+     */
+    @Nullable
+    private final CompoundNBT compound;
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowResourceList.java:84]: Why do we need this field?

File: src/main/java/com/minecolonies/coremod/client/gui/WindowResourceList.java [hunk 2]
```diff
@@ -77,6 +124,43 @@ public WindowResourceList(final int colonyId, final BlockPos buildingPos)
...
+    private void pullResourcesFromWarehouse()
+    {
+        if (warehousePos == null)
+        {
+            return;
+        }
+
+        Map<Item, BuildingBuilderResource> resourceMap = resources.stream().collect(Collectors.toMap(ItemStorage::getItem, v -> v));
+
+        warehouseSnapshot = new HashMap<>();
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowResourceList.java]: can't we always just initiate this with an empty map?

**Review by Raycoms — CHANGES_REQUESTED** [review]
Some small questions

File: src/main/java/com/minecolonies/coremod/client/gui/WindowResourceList.java (hunk 2 above)
Comment (Thodor12) [reply src/main/java/com/minecolonies/coremod/client/gui/WindowResourceList.java]: In the constructor it attempts to load the snapshot if any NBT data is present, else it's an empty hashmap, however this resets the snapshot in order to clear out a potential existing snapshot to then proceed to populate it with new data.

File: src/main/java/com/minecolonies/coremod/client/gui/WindowResourceList.java (hunk 1 above)
Comment (Thodor12) [reply src/main/java/com/minecolonies/coremod/client/gui/WindowResourceList.java:84]: The snapshot is calculated inside of this class, so we have to pass along the compound in order to save the snapshot.
It was either passing along the compound or the resource scroll itself.

Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/client/gui/WindowResourceList.java:84]: yeah, we pass it along, but do we need it as a field. couldn't see where we needed the field where we didn't already pass it as a param

File: src/main/java/com/minecolonies/coremod/client/gui/WindowResourceList.java (hunk 2 above)
Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/client/gui/WindowResourceList.java]: ah kk

File: src/main/java/com/minecolonies/coremod/client/gui/WindowResourceList.java (hunk 1 above)
Comment (Thodor12) [reply src/main/java/com/minecolonies/coremod/client/gui/WindowResourceList.java:84]: I don't think so as this saving is not directly called through the constructor.
It first goes through `onOpened` which then proceeds to call `pullResourcesFromWarehouse`, that is where the saving happens.

Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/client/gui/WindowResourceList.java:84]: I double checked the code, the compound is only used in saving, but do we need that data there at all? Shouldn't we write completely fresh data anyway?

Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/client/gui/WindowResourceList.java:84]: technically, we could give the message only the warehouse snapshot data, colony id and building id (which are needed extra anyway to identify)

**Review by Raycoms — DISMISSED** [review]

**Review by someaddons — DISMISSED** [review]

**Review by marchermans — DISMISSED** [review]

**Review by Raycoms — APPROVED** [review]

**Review by marchermans — APPROVED** [review]

**Pull request merged**